### PR TITLE
Update rest identity service

### PR DESCRIPTION
**Pull request by jsonsivar**
* Save email in a way we can fetch by email after
 * Added `displayname` query
 * Added `getIdentities` query

NOTE: the test harness seem to be broken before these changes so asking permission to fix those after.

### CODE REVIEW

File: services/rest/src/identities.ts [hunk 1]
```diff
@@ -11,25 +11,37 @@ const identityDb = new IdentityModel(STAGE);
 const queryBy = {
   address: (str: string) => identityDb.getIdentityByAddress(str),
   username: (str: string) => identityDb.getIdentityByUsername(str),
+  email: (str: string) => identityDb.getIdentityByEmail(str),
 };
 
+declare type FilterKey = 'address' | 'email' | 'username';
```
Comment (perfectmak) [inline services/rest/src/identities.ts:17]:
```suggestion
type FilterKey = 'address' | 'email' | 'username';
```

Comment (jsonsivar) [reply services/rest/src/identities.ts]: Fixed in [f2019f3](https://github.com/FleekHQ/space-services/pull/36/commits/f2019f36d1bead6cd42ef229f113d6b9c4a288ab).

File: packages/models/src/identity/index.ts [hunk 2]
```diff
@@ -94,6 +96,27 @@ export class IdentityModel extends BaseModel {
...
+      'pk = :displayName';
+    const ExpressionAttributeValues: DocumentClient.ExpressionAttributeValueMap = {
+      ':displayName': `${dn}`,
+    };
+
+    const params = {
+      TableName: this.table,
+      KeyConditionExpression,
+      ExpressionAttributeValues,
+      IndexName: 'gs1',
```
Comment (studna) [inline packages/models/src/identity/index.ts:112]: is this correct index? I mean isn't that one used with gs1pk and gs1sk?

Comment (jsonsivar) [reply packages/models/src/identity/index.ts]: Good catch. Fixed in [8518135](https://github.com/FleekHQ/space-services/pull/36/commits/8518135f7d655c08c7b280567e1d37db344889d4).

**Pull request merged**